pom.xml

```diff
@@ -788,6 +808,21 @@
     </pluginManagement>
 
     <plugins>
+      <plugin>
+        <groupId>net.revelc.code.formatter</groupId>
+        <artifactId>formatter-maven-plugin</artifactId>
+        <configuration>
+          <configFile>../jboss-java-formatter.xml</configFile>
+        </configuration>
+        <executions>
+          <execution>
+            <id>java-format</id>
+            <goals>
+              <goal>format</goal>
```

Comment:
You could move this to `pluginManagement` to facilitate running it manually, but I really don't want this to run automatically as part of every build. I think there is an issue that if it modifies the files, then the resulting jar and sources jar will have sources that do not match what's in the git repository, which I think is a problem.

Also, it doesn't seem to support an actual verify mode so that it could work similarly to checkstyle, which currently does fail the build if the sources are not formatted properly, but it cannot make any changes to the files.

Reply:
The good thing about it is, that it keeps the code formatted properly. If we ensure we don't push code, which is not formatted properly, the code and the sources jar will match.
To ensure this, we should have a check on the PR job, which checks if no files were updated - we do it this way. If we do this, the problem should be gone.

Reply:
I have no problem running it as a check in PR, but I am still worried about running it by default. We can put the required configuration into `pluginManagement` and then call `mvn net.revelc.code.formatter:formatter-maven-plugin:format` from the PR.

I am tempted to make a PR for this plugin that does a `verify` without changing the files. This, for me, is better for an automated run. There is a `formatter:validate` goal, but it does not work as I would expect.

Reply:
OK, we can configure it in the PR job.